src/sdk/namespace_fs.js

```diff
@@ -2578,10 +2578,19 @@ class NamespaceFS {
         }
     }
 
+    /**
+     * _delete_path_dirs deletes all the paths in the hierarchy that are empty after a successful delete
+     * if dir is not empty it will stop at the first non empty dir
+     * if dir is also a directory object (disabled mode optimization - CONTENT_DIR xattr is on the directory) we stop the loop
+     * @param {String} file_path 
+     * @param {nb.NativeFSContext} fs_context 
+     */
     async _delete_path_dirs(file_path, fs_context) {
         try {
             let dir = path.dirname(file_path);
             while (dir !== this.bucket_path) {
+                const dir_stat = await nb_native().fs.stat(fs_context, dir);
```

Comment:
we have a lot of the same checks in different ways. I think we should start standardizing the way we handle content dir and paths in general. I have made a function _is_disabled_content_dir that does this check. maybe you can see if you can use it here. or maybe create a new function that is called both in here and in _is_disabled_content_dir. also I think its better to use stat_ignore_enoent, because if this directory is deleted by a different process we will get an exception

Reply:
no need for stat_ignore_enoent because error is handled in catch. using _is_disabled_content_dir is not good here. we can do path handle refactoring in different PR if needed